Source/DafnyLanguageServer/Language/DiagnosticErrorReporter.cs

```diff
@@ -95,25 +95,22 @@ public class DiagnosticErrorReporter : ErrorReporter {
           var postcondition = entryDocumentsource.Substring(range.StartToken.pos, rangeLength);
           message = $"This postcondition might not hold: {postcondition}";
         } else if (message == "Related location") {
-          // We can do better than that.
           var tokenUri = tokenForMessage.GetDocumentUri();
           if (tokenUri == entryDocumentUri) {
             message = FormatRelated(entryDocumentsource.Substring(range.StartToken.pos, rangeLength));
           } else {
             var fileName = tokenForMessage.GetDocumentUri().GetFileSystemPath();
+            message = "";
             try {
-              var file = File.OpenRead(fileName);
-              var toRead = new byte[rangeLength];
-              file.Read(toRead, range.StartToken.pos, rangeLength);
-              file.Close();
-              var read = toRead.ToString();
-              if (read != null) {
-                message = FormatRelated(read);
-              } else {
-                message = "Related location (could not read file " + fileName + ")";
-              }
+              var content = File.ReadAllText(fileName);
+              message = FormatRelated(content.Substring(range.StartToken.pos, rangeLength));
```

Comment:
I guess trading the previously optimized code for this simple but costly one is worth the fact that we don't need to deal with encodings. That's fair.

Reply:
I think all included files should be loaded into memory and available through some in memory filesystem, so then there wouldn't be any extra file read here.